tests/Doctrine/Tests/ORM/Functional/Ticket/GH6823Test.php

```diff
@@ -0,0 +1,140 @@
+        self::assertEquals('CREATE TABLE gh6823_user_tags (user_id VARCHAR(255) CHARACTER SET utf8mb4 NOT NULL COLLATE `utf8mb4_bin`, tag_id VARCHAR(255) CHARACTER SET latin1 NOT NULL COLLATE `latin1_bin`, INDEX IDX_596B1281A76ED395 (user_id), INDEX IDX_596B1281BAD26311 (tag_id), PRIMARY KEY(user_id, tag_id)) DEFAULT CHARACTER SET utf8 COLLATE `utf8_unicode_ci` ENGINE = InnoDB', $this->getLastLoggedQuery(5)['sql']);
+        self::assertEquals('CREATE TABLE gh6823_group (id VARCHAR(255) NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET ascii COLLATE `ascii_general_ci` ENGINE = InnoDB', $this->getLastLoggedQuery(4)['sql']);
+        self::assertEquals('CREATE TABLE gh6823_status (id VARCHAR(255) CHARACTER SET latin1 NOT NULL COLLATE `latin1_bin`, PRIMARY KEY(id)) DEFAULT CHARACTER SET koi8r COLLATE `koi8r_bin` ENGINE = InnoDB', $this->getLastLoggedQuery(3)['sql']);
+        self::assertEquals('ALTER TABLE gh6823_user ADD CONSTRAINT FK_70DD1774FE54D947 FOREIGN KEY (group_id) REFERENCES gh6823_group (id)', $this->getLastLoggedQuery(2)['sql']);
+        self::assertEquals('ALTER TABLE gh6823_user ADD CONSTRAINT FK_70DD17746BF700BD FOREIGN KEY (status_id) REFERENCES gh6823_status (id)', $this->getLastLoggedQuery(1)['sql']);
+        self::assertEquals('ALTER TABLE gh6823_user_tags ADD CONSTRAINT FK_596B1281A76ED395 FOREIGN KEY (user_id) REFERENCES gh6823_user (id)', $this->getLastLoggedQuery(0)['sql']);
+    }
+}
+
+/**
+ * @Entity
+ * @Table(name="gh6823_user", options={
+ *     "charset"="utf8mb4",
+ *     "collation"="utf8mb4_bin"
+ * })
+ */
+class GH6823User
+{
+    /**
+     * @var string
+     * @Id
+     * @Column(type="string")
+     */
+    public $id;
+
+    /**
+     * @var GH6823Group
+     * @ManyToOne(targetEntity="GH6823Group")
+     * @JoinColumn(name="group_id", referencedColumnName="id", options={"charset"="ascii", "collation"="ascii_general_ci"})
+     */
+    public $group;
+
+    /**
+     * @var GH6823Status
+     * @ManyToOne(targetEntity="GH6823Status")
+     * @JoinColumn(name="status_id", referencedColumnName="id", options={"charset"="latin1", "collation"="latin1_bin"})
+     */
+    public $status;
+
+    /**
+     * @var Collection<int, GH6823Tag>
+     * @ManyToMany(targetEntity="GH6823Tag")
+     * @JoinTable(name="gh6823_user_tags", joinColumns={
+     *   @JoinColumn(name="user_id", referencedColumnName="id", options={"charset"="utf8mb4", "collation"="utf8mb4_bin"})
+     * }, inverseJoinColumns={
+     *   @JoinColumn(name="tag_id", referencedColumnName="id", options={"charset"="latin1", "collation"="latin1_bin"})
+     * })
```

Comment:
I want this to be tested for both Annotations and PHP 8 Attributes. 

I was thinking of the following:
* Add both Annotations and Attributes
* Create a `@dataProvider` that returns `annotations` and `attributes`
* When `annotations` configure the metadata factory to only look at annotations
* When `attributes` configure the metadata factory to only look at attributes

For both configurations, the output SQL should be the same.

Why do I want this? Because there are subtle differences with the example above. With PHP Attributes, you need to use `#[InverseJoinColumn]` which is an Attribute only. 

@greg0ire Do you have an idea on how to do this? Or is this already happening somewhere in the tests?

Reply:
Decided to just test this here:
tests/Doctrine/Tests/ORM/Mapping/AttributeReaderTest.php